src/shared/Specimen.ts

```diff
@@ -52,23 +87,53 @@ export class Specimen {
     get sequence(): SequenceInterface<GenericParamDescription> {
         return this._sequence
     }
+    /**
+     * Assigns a new name to this specimen
+     */
+    set name(name: string) {
+        this._name = name
+    }
     /**
      * Assigns a new visualizer to this specimen and updates its sequence
      * to match the specimen. It also ensures this visualizer inhabits
      * the correct HTML element and begins to render.
+     * @param visualizerKey the key of the desired visualizer's export module
      */
-    set visualizer(visualizer: VisualizerInterface<GenericParamDescription>) {
-        this._visualizer = visualizer
+    set visualizerKey(visualizerKey: string) {
+        this._visualizerKey = visualizerKey
+        this._visualizer = new vizMODULES[visualizerKey].visualizer(
+            this._sequence
+        )
         if (this.isSetup) this.setup(this.location!)
     }
     /**
      * Assigns a new sequence to this specimen and updates the visualizer
      * to reflect this change in the render.
+     * @param specimenKey the key of the desired sequence's export module
      */
-    set sequence(sequence: SequenceInterface<GenericParamDescription>) {
-        this._sequence = sequence
+    set sequenceKey(sequenceKey: string) {
+        this._sequenceKey = sequenceKey
+        type SeqIntf = SequenceInterface<GenericParamDescription>
+        if (seqMODULES[sequenceKey].kind === SequenceExportKind.FAMILY)
+            this._sequence = new (seqMODULES[sequenceKey]
+                .sequenceOrConstructor as SequenceConstructor)(0)
+        else
+            this._sequence = seqMODULES[sequenceKey]
+                .sequenceOrConstructor as SeqIntf
         this.visualizer.view(this.sequence)
     }
+    /**
+     * Exists for redundancy, same as set visualizerKey()
+     */
+    set visualizer(visualizerKey: string) {
+        this.visualizerKey = visualizerKey
+    }
+    /**
+     * Exists for redundancy, same as set sequenceKey()
+     */
+    set sequence(sequenceKey: string) {
+        this.sequenceKey = sequenceKey
+    }
```

Comment:
Why do we want redundant names for these two setters? Redundancy in interfaces is in general a bad thing, not a good thing. Please remove unless there's a strong argument for having these setters in addition to visualizerKey and sequenceKey.